odoo/addons/test_new_api/tests/test_ui.py

```diff
@@ -14,3 +15,20 @@ def test_01_admin_widget_x2many(self):
         # too small or there are too many items preceding it in the tests menu
         self.start_tour("/web#action=test_new_api.action_discussions",
             'widget_x2many', step_delay=100, login="admin", timeout=120)
+
+
+class TestUiTranslation(odoo.tests.HttpCase):
+
+    @mute_logger('odoo.sql_db', 'odoo.http')
+    def test_01_sql_constraints(self):
+        # Raise an SQL constraint and test the message
+        # As the transaction is rollbacked at insert and a new cusor is opened,
+        # can not test that the message is translated (load_module_terms is also)
+        # rollbacked. Test individually the external id and loading of translation.
+        self.env['ir.translation'].load_module_terms(['test_new_api'], ['fr_FR'])
+        constraint = self.env.ref('test_new_api.constraint_test_new_api_category_positive_color')
+        message = constraint.with_context(lang='fr_FR').message
+        self.assertEqual(message, "La couleur doit être une valeur positive !")
+
+        self.start_tour("/web#action=test_new_api.action_categories",
+            'sql_constaint', login="admin")
```

Comment:
Why keeping the tour if it does not work for translations?

Reply:
to check the error is correctly violated